File: 0104-MaximumDepthOfBinaryTree/Solution.java
import java.util.*;

public class Solution {

    static int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        } else {
            int leftDepth = maxDepth(root.left);
            int rightDepth = maxDepth(root.right);
            return 1 + Math.max(leftDepth, rightDepth);
        }
    }

    static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public static void main(String[] args) {
        List<TreeNode> inputs = new ArrayList<>();
        List<Integer> expected = new ArrayList<>();

        inputs.add(null);
        expected.add(0);

        inputs.add(new TreeNode(1));
        expected.add(1);

        inputs.add(new TreeNode(1,
                       null,
                       new TreeNode(1)
                   )
        );
        expected.add(2);

        inputs.add(new TreeNode(1,
                       new TreeNode(1),
                       new TreeNode(1,
                           new TreeNode(1),
                           new TreeNode(1)
                       )
                   )
        );
        expected.add(3);

        for (int i = 0; i < inputs.size(); i++) {
            System.out.println(expected.get(i));
            System.out.println(maxDepth(inputs.get(i)));
            System.out.println();
        }
    }

}

File: 0104-MaximumDepthOfBinaryTree/solution.py
from __future__ import annotations


class Solution:

    def max_depth(self, root: TreeNode) -> int:
        if not root:
            return 0
        else:
            left_depth = self.max_depth(root.left)
            right_depth = self.max_depth(root.right)
            return 1 + max(left_depth, right_depth)


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


if __name__ == "__main__":
    s = Solution()

    inputs: List[TreeNode] = []
    expected: List[int] = []

    inputs.append(None)
    expected.append(0)

    inputs.append(TreeNode(1))
    expected.append(1)

    inputs.append(
            TreeNode(1,
                None,
                TreeNode(1)))
    expected.append(2)

    inputs.append(
            TreeNode(1,
                TreeNode(1),
                TreeNode(1,
                    TreeNode(1),
                    TreeNode(1))))
    expected.append(3)

    for inp, exp in zip(inputs, expected):
        print(exp, s.max_depth(inp), '', sep='\n')

